File: src/main/java/com/example/controller/HelloController.java
package com.example.controller;

import io.micronaut.http.annotation.*;

@Controller("/hello")
public class HelloController {

    @Get("/")   // GET /hello
    public String index() {
        return "Hello from Micronaut!";
    }


    @Get("/{name}")   // GET /hello/{name}
    public String greet(String name) {
        return "Hello, " + name + "!";
    }
}

File: src/main/kotlin/com/example/kotlin/Hellocontroller.kt
package com.example.controller

import io.micronaut.http.annotation.*

@Controller("/hello")
class HelloController {

    @Get("/")   // GET /hello
    fun index(): String = "Hello from Micronaut!"

    @Get("/{name}")   // GET /hello/{name}
    fun greet(name: String): String = "Hello, $name!"
}
